lib/rubygems/uninstaller.rb

```diff
@@ -376,4 +380,19 @@ def safe_delete(&block)
 
     raise e
   end
+
+  private
+
+  def announce_deletion_of(spec)
+    name = spec.full_name
+    say "Successfully uninstalled #{name}"
+    if spec.default_gem?
+      say(
+        "#{name} is a default gem. A user copy has been deleted, but the " \
+          "default copy cannot be deleted. Please see " \
+          "https://github.com/rubygems/rubygems/issues/2112 which explains " \
+          "the difference between (and precedence of) user and default copies."
+      )
+    end
```

Comment:
I think this is incorrect, because we partition the list of initial candidates between default specs and regular specs here:

https://github.com/rubygems/rubygems/blob/6a5a1a4af23526e24f21b558517bb8948a6e4ff5/lib/rubygems/uninstaller.rb#L98-L101

And we don't even try to uninstall default gems, so `spec.default_gem?` will always be `false`?

Regarding the message, I would keep the message as is for the standard case, and change it to something simpler like

"There was a regular copy and a default copy of #{name}. The regular copy was successfully uninstalled, but the default copy was left around because default gems can't be removed."

Reply:
Hopefully resolved by https://github.com/rubygems/rubygems/pull/4739/commits/ff8c129a9f0ceca27ad279f9b30ba4b8631124b5